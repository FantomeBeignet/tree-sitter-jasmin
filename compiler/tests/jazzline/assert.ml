open Common

let () =
  parse_and_print "assert.jazz"  "mul";
  parse_and_print "assert.jazz"  "mul2";
  parse_and_print "assert.jazz"  "mul4"
