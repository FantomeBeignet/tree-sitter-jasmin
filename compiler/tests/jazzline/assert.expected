proc mul(z_167@uint64, x_150@uint64, y_151@uint64) =
{
 true && true
}
assert true && (x_150@uint64 = (const 64 (2)));
mull TMP_____178@uint64 z_167@uint64 x_150@uint64 y_151@uint64;
assert true && (z_167@uint64 = ((const 64 (2)) * y_151@uint64));
{
 true && true
}
 proc mul4(y_229@uint64, y1_207@uint64) =
{
 true && true
}
mov x_226@uint64 (2)@uint64;
assert true && (x_226@uint64 = (const 64 (2)));
mull TMP_____230@uint64 y_227@uint64 x_226@uint64 y1_207@uint64;
assert true && (y_227@uint64 = ((const 64 (2)) * y1_207@uint64));
mov x_228@uint64 (2)@uint64;
assert true && (x_228@uint64 = (const 64 (2)));
mull TMP_____231@uint64 y_229@uint64 x_228@uint64 y_227@uint64;
assert true && (y_229@uint64 = ((const 64 (2)) * y_227@uint64));
assert true && (y_229@uint64 = ((const 64 (4)) * y1_207@uint64));
{
 true && true
}
 